### DISCUSSION TITLE
tag parsing error

### DISCUSSION THREAD

**Discussion by jalanh11640** [microsoft/vscode-cpptools#10188] [Q&A]
Whenever I search for workplace symbols (cmd-t on my mac) I get 'No matching workspace symbols'.  Some google searching suggested resetting my intellisense database.  When I do that, I see the following in my C/C++ Log (loggingLevel set to 'Debug')

Parsing remaining files...
  tag parsing file: <full file path>
    tag parsing error (this can be ignored unless symbols can't be found):
  <Repeated for 118 more files>
  Parsing: 119 files(s) processed
Done parsing remaining files.

Further googling seemed to indicate I should see more detailed information about what went wrong.  Here's the info about the versions I'm using

Vscode: 1.72.2 (Mac app)
Remote-Ssh: 0.92.0
c/c++ extension 1.13.4 (installed on a remote Red Hat linux box) same result with pre-release version

How can I figure out what the problem/issue is?
Thanks!

**Comment by sean-mcmanus** [comment]
@jalanh11640 You can set the `C_Cpp.loggingLevel` to "Debug" to see more information on which files are tag parsed. UPDATE: Oops, I didn't see you already have that setting.

It sounds like you might be hitting a tag parser error that is causing symbols to not be available. Are you able to provide any sample repro code?

**Reply by sean-mcmanus** [reply]
@jalanh11640 Yeah, with 1.13.4 (pre-release) you should see additional error info like `  line 695 (col. 15): error: `.

**Reply by jalanh11640** [reply]
> @jalanh11640 You can set the `C_Cpp.loggingLevel` to "Debug" to see more information on which files are tag parsed. UPDATE: Oops, I didn't see you already have that setting.
> 
> It sounds like you might be hitting a tag parser error that is causing symbols to not be available. Are you able to provide any sample repro code?

I'm working on paring down my configuration/code so that I can post it here.

**Reply by jalanh11640** [reply]
Here is my C/C++ log after resetting the intellisense database.  In case it points to something, parsing the 19,942 files takes approximately 9m45s.

cpptools/resetDatabaseReady
Database safe to open
Code browsing service initialized
Discovering files...
  Processing folder (recursive): /usr/include/
  Processing folder (recursive): /usr/lib/gcc/x86_64-redhat-linux/4.8.2/include/
  Processing folder (recursive): /usr/local/include/
  Processing folder (recursive): /home/harrisja/parser/
  Discovering files: 19942 file(s) processed
  0 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  tag parsing file: /home/harrisja/parser/TelemetryList.cpp
    tag parsing error (this can be ignored unless symbols can't be found):
  tag parsing file: /home/harrisja/parser/TelemetryList.hpp
    tag parsing error (this can be ignored unless symbols can't be found):
  Parsing: 2 files(s) processed
Done parsing remaining files.

Here is my c_cpp_properties.xml
{
        "configurations": [
        {
            "name": "SAMAXgnuDEBUG",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "compilerPath": "/bin/g++",
            "cppStandard": "c++03",
            "intelliSenseMode": "gcc-x86",
            "browse": {
                "limitSymbolsToIncludedHeaders": true,
                "path": [
                    "${workspaceFolder}"
                ]
            }
        }
    ],
    "version": 4
}

Here's my code-workspace file.
{
	"folders": [
		{
			"path": "."
		}
	],
	"settings": {
		"C_Cpp.loggingLevel": "Debug"
	}
}

I'll try to post some code tomorrow that exhibits this behavior.

Thanks

**Reply by jalanh11640** [reply]
> @jalanh11640 Yeah, with 1.13.4 (pre-release) you should see additional error info like ` line 695 (col. 15): error:`.

Interesting, I don't see any error info like that.  Is it possible an extension update didn't work as expected and the extension is in some sort of wonky state?  Last week I did blow away my ~/.vscode-server directory on the remote machine and re-installed the remote extensions..  not sure if there's something that would need to be cleared out on the mac side as well.

I've pared down my code to a default constructor and destructor both with empty implementations (in the cpp) and still get the issue where there are no workspace or editor symbols.

**Reply by sean-mcmanus** [reply]
It's possible there is some failure in creating the necessary temp files that the additional error logging is written to.

1.13.5 should have the Create Declaration / Definition feature in the right-click menu for a C/C++ document.

Are you able to provide some sample repro code?

**Reply by jalanh11640** [reply]
TelemetryList.hpp:
class CTelemetryRequestList
{
    public:
        CTelemetryRequestList();
        ~CTelemetryRequestList();
};

TelemetryList.cpp:
#include "TelemetryList.hpp"

CTelemetryRequestList::CTelemetryRequestList()
{
}

CTelemetryRequestList::~CTelemetryRequestList()
{
}

**Reply by sean-mcmanus** [reply]
Yeah, that is strange -- I don't repro the issue. I don't know what could cause a failure like this. Does it stop reproing with something simpler like `class CTelemetryRequestList;`? Also, you're getting the tag parse error with the header and source file?

**Reply by jalanh11640** [reply]
Yes, I'm getting the tag parse error with the header and source.  If I just have 'class CTelemetryRequestList;' in my header and an empty cpp I still get the tag parse errors for both files.

**Reply by sean-mcmanus** [reply]
You might want to try running the "C/C++: Reset IntelliSense Database" command from the Command Palette, but normally that would only fix database errors.

**Comment by jalanh11640** [comment]
Thank you for taking the time to look into this.  When my work schedule allows, I'm going to uninstall/clean EVERYTHING (Mac and remote) and see if that helps with anything.

At this point, vscode is mostly useable.

Thanks again

**Reply by jalanh11640** [reply]
Interesting data point..  I remoted into a different Linux box (one I'd never remoted into with vscode).  The parser found most all files.  There were a number of '/usr/include/boost/** files that it had problems with.. in that case it showed additional error information.

Jeff

**Reply by sean-mcmanus** [reply]
Yeah, that sounds like the expected behavior. It seems like your other Linux machine is causing some low level functionality in our tag parser to fail, but we don't know what that could be yet (apparently the error may also be causing our error failure reporting mechanism to fail or it's failing at a point in which our existing error reporting isn't designed to be able to report detailed errors). I don't recall other reports of this so it seems like something unusual.

**Reply by sean-mcmanus** [reply]
It sounds like another user may have hit this at https://github.com/microsoft/vscode-cpptools/issues/10224
